### PR TITLE
e2e tests: skeleton structure

### PR DESCRIPTION
**Pull request by baude**
this is a table-setting for greating things.  we can use this as a basic structure for creating e2e tests for libhvee.

### CODE REVIEW

File: pkg/test/e2e/basic_test.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+	BeforeEach(func() {
+		// setup stuff can go here
+	})
+
+	AfterEach(func() {
+		// teardown stuff can go here
+	})
+
+	FIt("create, start, stop, remove", func() {
```
Comment (ashley-cui) [inline pkg/test/e2e/basic_test.go:35]:
```suggestion
	It("create, start, stop, remove", func() {
```

File: pkg/test/e2e/libhvee_test.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+func TestLibhvee(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Libhvee Suite")
+}
+
+func get(endpoint string) ([]byte, error) {
+	resp, err := http.Get(endpoint)
+	if err != nil {
+		// handle error
```
Comment (ashley-cui) [inline pkg/test/e2e/libhvee_test.go:32]:
```suggestion
		// handle error
		return err
```

File: pkg/test/e2e/vm_test.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+func (t *testVM) stopAndRemove() error {
+	if err := t.vm.StopWithForce(); err != nil {
+		fmt.Println(err)
+		return err
+	}
+	err := t.vm.Remove(t.config.DiskPath)
+	if err != nil {
+		fmt.Println(err)
+	}
+	return err
```
Comment (ashley-cui) [inline pkg/test/e2e/vm_test.go:42]:
```suggestion
	return t.vm.Remove(t.config.DiskPath)
```

**Review by ashley-cui — COMMENTED** [review]
Some small nits, otherwise initial review LGTM

**Comment by n1hility** [comment]
LGTM. 

One thing I wonder for the future is would it make sense to have a section or chunk of tests that dont require a disk image, or just use a mock/ zeroed non bootable image? That way once this gets into CI you can have a number of common failures failing fast before waiting on a download, or if there is a network / fedora blip some of the tests still work. Just a thought and more of a nice to have.

**Comment by baude** [comment]
> One thing I wonder for the future is would it make sense to have a section or chunk of tests that dont require a disk image, or just use a mock/ zeroed non bootable image?

Yes, I can most certainly imagine this ... i tried to write things as open as possible so we have plenty of wiggle room

**Pull request merged**